### ISSUE TITLE
Takes considerable time on a tone of objects.

### ISSUE THREAD

**Issue by Kozzren** [Weisl/simple_renaming_panel#193]
So i use this in conjunction with naming many objects at once for with track creation for assetto corsa, and ive noticed on 1k + objects selected it really chugs a long and takes a good amount of time to rename all of them. Specially si the case when rename massive groups of trees.

Here are my pc specs for future reference:
CPU - Ryzen 9 5950x
GPU - Radeon 7900 XTX Msi Gaming Trio Classic
RAM - 64gb @ 3200hz

**Comment by Weisl** [comment]
Does this also happen with Use Selection Order disabled? 
<img width="328" alt="selectionOrder" src="https://github.com/user-attachments/assets/f0bdc2a0-5d60-46f1-93d7-3cbcb9ecb8e2">

**Comment by Kozzren** [comment]
Yes it does but not as long as it seem. I just tried it on another map with roughly 1500 trees, was faster but still a bit slow

**Comment by Derjyn** [comment]
I know it's been a year since this issue was posted (_looks like some housekeeping is in order_), but I'm curious what is considered "considerable time". I have several projects with thousands of objects that could use some organization (_logical renaming_), so wouldn't mind having some ground-truth values for what's standard processing time.

I'm a big fan of not spinning my tires waiting on something - if I'm waiting for something that's actually crashed/failed, I want to be aware so I can move forward rather than twiddling my thumbs. I've already tested with a few hundred objects, and it zipped right along, taking several seconds. I have a mid-end rig, compared to @Kozzren, so curious if their idea of considerable time is different than mine. If it takes 30 seconds to rename 2000 items, that's quite alright with me. Much faster than doing it manually!
